modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/MuleExtensionModelProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.resources;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.util.IOUtils;
+import org.mule.runtime.extension.api.persistence.ExtensionModelJsonSerializer;
+
+/**
+ * Utility class to access the {@link ExtensionModel} definition for Mule's Runtime
+ *
+ * @since 4.0
+ */
+public class MuleExtensionModelProvider {
+
+  private static final String MODEL_JSON = "META-INF/mule-extension-model.json";
+  private static final ExtensionModel EXTENSION_MODEL = new ExtensionModelJsonSerializer(false)
```

Comment:
Agree in that is no ideal, but this change is better than what we had and should be more than enough since this is only used by tooling, which already works with all deserialized plugins.
It sounds simple to provide this model without using the json, it implies creating the whole declaration manually, going through the factory, etc.
Not using the Json sounds like a different ticket to me. We can always change later how we create the model.